app/js/streaming/StreamController.js

```diff
@@ -298,12 +298,33 @@
             self.notify(MediaPlayer.dependencies.StreamController.eventList.ENAME_STREAMS_COMPOSED);
         },
 
+        onTimeSyncAttemptCompleted = function (e) {
+            var self = this,
+                manifest;
+
+            if (!e.error) {
+                // adjust the loadedTime of the manifest by the offset between
+                // the network time and the local time. this is because loadedTime
+                // is used in a number of places to work out availability etc.
+                manifest = self.manifestModel.getValue();
```

Comment:
Fixed in e40fa79. Revisiting this has simplified the solution I think, so thanks for the comments and sorry for wasting your time with the first iteration!